newIDE/app/src/MainFrame/EditorContainers/HomePage/TeamSection/NewTeamGroupNameField.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+
+import * as React from 'react';
+import { Line } from '../../../../UI/Grid';
+import IconButton from '../../../../UI/IconButton';
+import CheckIcon from '../../../../UI/CustomSvgIcons/Check';
+import CrossIcon from '../../../../UI/CustomSvgIcons/Cross';
+import {
+  shouldValidate,
+  shouldCloseOrCancel,
+} from '../../../../UI/KeyboardShortcuts/InteractionKeys';
+import TextField from '../../../../UI/TextField';
+import { Trans, t } from '@lingui/macro';
+
+type Props = {|
+  onValidateGroupName: ({| name: string |}) => Promise<void>,
+  onDismiss: () => void,
+|};
+
+const NewTeamGroupNameField = ({ onValidateGroupName, onDismiss }: Props) => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [errorText, setErrorText] = React.useState<?React.Node>(null);
+  const [groupName, setGroupName] = React.useState<string>('');
+
+  const onFinishEditingName = React.useCallback(
+    async () => {
+      const cleanedGroupName = groupName.trim();
+      if (!cleanedGroupName) {
+        setErrorText(<Trans>Group name cannot be empty.</Trans>);
+        return;
+      }
+      setIsLoading(true);
+      try {
+        await onValidateGroupName({ name: cleanedGroupName });
+        setGroupName('');
+        setErrorText(null);
+        onDismiss();
+      } catch (error) {
+        console.error(error);
+        setErrorText(
+          <Trans>
+            An error occurred while creating the group. Please try again later.
+          </Trans>
+        );
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [onValidateGroupName, groupName, onDismiss]
+  );
+
+  const onCancelEditingName = React.useCallback(
+    () => {
+      setGroupName('');
+      setErrorText(null);
+      onDismiss();
+    },
+    [onDismiss]
+  );
+
+  return (
+    <Line noMargin>
+      <TextField
+        type="text"
+        maxLength={50}
+        value={groupName}
+        disabled={isLoading}
+        margin="dense"
+        onChange={(e, newName) => {
+          setGroupName(newName);
+        }}
+        autoFocus="desktopAndMobileDevices"
+        translatableHintText={t`New group name`}
+        errorText={errorText}
+        onKeyUp={event => {
+          if (shouldValidate(event)) {
+            onFinishEditingName();
+          } else if (shouldCloseOrCancel(event)) {
+            event.stopPropagation();
+            event.preventDefault();
```

Comment:
you need those 2 event actions? one is not enough? (I don't know, just asking cause it's the first time I see both used at the same time)

Reply:
I'll dig into it